dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/S3Utils.java

```diff
@@ -175,21 +176,33 @@ public String getFileName(ResourceType resourceType, String tenantCode, String f
     }
 
     @Override
-    public void download(String tenantCode, String srcFilePath, String dstFile, boolean deleteSource, boolean overwrite) throws IOException {
+    public void download(String tenantCode, String srcFilePath, String dstFilePath, boolean deleteSource, boolean overwrite) throws IOException {
+        File dstFile = new File(dstFilePath);
+        if (dstFile.exists()) {
+            if (overwrite) {
+                if (dstFile.isDirectory()) {
+                    Files.delete(dstFile.toPath());
+                }
+            } else {
+                logger.info("the destination path {} already exists, download operation will be ignored", dstFilePath);
+                return;
```

Comment:
We should remove this return.

Reply:
> We should remove this return.

If so, It will result in unnecessary requests for s3 resources

Reply:
It is useful when the user modifies the contents of the file and reruns the task.

Reply:
> It is useful when the user modifies the contents of the file and reruns the task.
when the user modifies the contents of the file，what is the download param overwrite value

Reply:
> It is useful when the user modifies the contents of the file and reruns the task.


Do you mean that the destination file should be overwritten no matter what the overwrite parameter is?

Reply:
> Currently `overwrite` is always true. I'm not sure why there is an overwrite parameter at present. And i think we should do this like `org.apache.dolphinscheduler.common.utils.HadoopUtils#copyHdfsToLocal` using `if (!dstPath.getParentFile().exists() && !dstPath.getParentFile().mkdirs()) { return false; }`

The download method of interface StorageOperate has no return value. The return value of org.apache.dolphinscheduler.common.utils.HadoopUtils#copyHdfsToLocal is not processed in download. Are you sure you want to return a boolean value in S3 download?

Reply:
> Currently `overwrite` is always true. I'm not sure why there is an overwrite parameter at present. And i think we should do this like `org.apache.dolphinscheduler.common.utils.HadoopUtils#copyHdfsToLocal` using `if (!dstPath.getParentFile().exists() && !dstPath.getParentFile().mkdirs()) { return false; }`


If the parameter is always true, then it will not cause rerun error, should we keep our processing logic?

Reply:
What i mean is if there is no exception in `dstPath.getParentFile().exists() && !dstPath.getParentFile().mkdirs()`. We should always download s3 files to the local path.

Reply:
> What i mean is if there is no exception in `dstPath.getParentFile().exists() && !dstPath.getParentFile().mkdirs()`. We should always download s3 files to the local path.

I have dropped overwrite related processing